datalad/support/annexrepo.py

```diff
@@ -229,25 +257,24 @@ def annex_proxy(self, git_cmd):
 
         Returns:
         --------
-        str containing stdout of the command
+        (stdout, stderr)
+            output of the command call
         """
 
         cmd_str = "git annex proxy -- %s" % git_cmd
-        # TODO: By now git_cmd is expected to be string. Figure out how to deal with a list here.
+        # TODO: By now git_cmd is expected to be string.
+        # Figure out how to deal with a list here.
 
         if not self.is_direct_mode():
-            lgr.warning("annex_proxy() called in indirect mode: %s" % git_cmd)
-            raise CommandNotAvailableError(cmd=cmd_str, msg="Proxy doesn't make sense if not in direct mode.")
-        try:
-            out = self.cmd_call_wrapper(cmd_str, shell=True, cwd=self.path)
-        except CommandError, e:
-            if "Unknown command" in e.stderr:
-                raise CommandNotAvailableError(cmd=cmd_str, msg=e.msg,
-                                               stderr=e.stderr, stdout=e.stdout)
-            else:
-                raise e
-        return out
-
+            lgr.warning("annex_proxy() called in indirect mode: %s" % cmd_str)
+            raise CommandNotAvailableError(cmd=cmd_str,
+                                           msg="Proxy doesn't make sense"
+                                               " if not in direct mode.")
+        import shlex
```

Comment:
this one is a standard library so better be imported at the top of the file once and for all